src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -20,21 +21,21 @@ namespace Microsoft.VisualStudio.TestPlatform.Client
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Utilities;
     using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers;
     using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers.Interfaces;
-    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Host;
-    using Microsoft.VisualStudio.TestPlatform.Common.Hosting;
 
     /// <summary>
     /// Implementation for TestPlatform
     /// </summary>
     public class TestPlatform : ITestPlatform
     {
-        private IFileHelper fileHelper;
+        private readonly TestRuntimeProviderManager testHostProviderManager;
+
+        private readonly IFileHelper fileHelper;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="TestPlatform"/> class.
         /// </summary>
-        public TestPlatform() : this(new TestEngine(), new FileHelper())
+        public TestPlatform() : this(new TestEngine(), new FileHelper(), TestRuntimeProviderManager.Instance)
```

Comment:
yes, If Taef wants to be the test runtime(i.e replace testhost), then they can provide their runtime whose job is to understand how to launch TAEF, once that is discovered, Taef can continue to host our crossplat engine to do the communication, & running test work, or they can implement their own communication framework which understands the protocol send across wire.